src/libraries/System.Console/src/System/IO/StdInReader.cs

```diff
@@ -171,42 +185,121 @@ private bool ReadLineCore(bool consumeKeys)
                     {
                         return false;
                     }
+                    // Alt + B/F is what macOS will send in place of Alt + Left/Right
+                    else if ((keyInfo.Key == ConsoleKey.B || keyInfo.Key == ConsoleKey.LeftArrow) && keyInfo.Modifiers == ConsoleModifiers.Alt)
```

Comment:
It would probably make sense to only capture Alt+B/F on MacOS platforms. Try using the new `System.OperatingSystem.IsMacOs()` method.

Reply:
Nit: would personally put modifier check before the key check.